code/src/components/Counter.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react'
+import { useSelector } from 'react-redux'
+import 'components/Counter.css'
+
+
+export const Counter = () => {
+
+    const items = useSelector((state) => state.tasks.items)
+    const tasksDone = items.filter((item) => item.needsMore)
```

Comment:
Since you're only using the length of the array, I'd recommend returning that directly from the selector:
(I know I didn't show this in the codealong, my bad)

for example:
`const items = useSelector((state) => state.tasks.items.length)`

When you do it like you have, this Counter component will re-render whenever the array of tasks changes (for example toggling a task's completion), but if you return the length from the selector, it'll only re-render when the length of the array changes. Small optimisation, but good to keep in the back of your head if you're working on a bigger project where you want to avoid unnecessary re-renders.